### PR TITLE
Remove excess exception logging for minifier, cleanup backend failure exception format

### LINKED ISSUE #1376: Minifier when run on benchmark scripts hides the minified script after many lines of scrollback
The backend failure gets printed a bazillion times, pushing the log line that says where the repro was written off the screen. Make this more user friendly.

### PR DESCRIPTION
**Pull request by mlazos**
Fixes https://github.com/pytorch/torchdynamo/issues/1376

Ensures exceptions are printed only in one place, once. 

implements some of the ideas from https://github.com/pytorch/pytorch/issues/93571
- Attaches a field to the exception which indicates that it's minified, a usage message is printed if this field is present

cc @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @jansel @lezcano @fdrocha

### CODE REVIEW

File: torch/_dynamo/debug_utils.py [hunk 1]
```diff
@@ -853,9 +853,8 @@ def dynamo_minifier_backend(gm, example_inputs, compiler_name):
     except Exception as exc:
         orig_failure = str(exc)
         log.warning(
-            "Compiled Fx GraphModule failed with following error. Starting minifier."
+            "Compiled Fx GraphModule failed. Creating script to minify the error."
```
Comment (anijain2305) [inline torch/_dynamo/debug_utils.py:856]: Will we see the failure at all? I think today, we see multiple copies of the same exception. Just confirming that we should have atleast one.

**Review by anijain2305 — APPROVED** [review]

File: torch/_dynamo/debug_utils.py (hunk 1 above)
Comment (mlazos) [reply torch/_dynamo/debug_utils.py:856]: Yeah we'll see it once, it's bubbled up to the top level in eval_frame.py

Comment (anijain2305) [reply torch/_dynamo/debug_utils.py:856]: Perfect

**Pull request closed without merging**